IdentityCore/src/controllers/broker/ios/MSIDBrokerInteractiveController.m

```diff
@@ -288,7 +288,7 @@ + (BOOL)completeAcquireToken:(nullable NSURL *)resultURL
        brokerResponseHandler:(nonnull MSIDBrokerResponseHandler *)responseHandler
 {
     // sourceApplication could be nil, we want to return early if we know for sure response is not from broker
-    if (sourceApplication && ![self isResponseFromBroker:sourceApplication])
+    if (!sourceApplication || (sourceApplication && ![self isResponseFromBroker:sourceApplication]))
```

Comment:
How would this work if sourceApplication is nil? (This happens every time when app is non-Microsoft signed)

Reply:
Thanks for clarifying, I was confused by the comment in the code. In this case, ```!sourceApplication``` is a valid case, and should be able to process with the rest correct?

Reply:
Updated the PR was we discuseed